src/view/components/AddEntry/index.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import { Content, List, ListItem, Text, Grid, Row, Col, H1, Button, Input, Form, Item, Label } from 'native-base';
+import * as moment from 'moment';
+
+import { Farmer } from '../../../store/modules/farmer/types';
+import { Dairy as MilkEntry } from '../../../store/modules/dairy/types';
+
+import { Route } from '../../navigation/navigator';
+import { MapStateToProps, MapDispatchToProps, connect } from 'react-redux';
+import navActions from '../../../store/modules/nav/actions';
+import { InjectedFabProps } from '../../hoc/PageComposer/FabPage/index';
+import Composer from '../../hoc/PageComposer/index';
+import { State } from '../../../store/types';
+import dairyActions from '../../../store/modules/dairy/actions';
+
+import Styles from './style';
+
+interface OwnPropsType {
+  farmer: Farmer;
+}
+
+interface DispatchPropsType {
+  navigate(route: Route): void;
+  goBack(): void;
+  createDairy(newEntry: MilkEntry): void;
+}
+
+interface StorePropsType {
+}
+
+type NestedPropsType = StorePropsType & DispatchPropsType & OwnPropsType;
+
+/** AddEntry PropsType */
+type PropsType = InjectedFabProps & NestedPropsType;
+
+interface OwnStateType {
+  volume: string;
+  quality: string;
+  costPerUnit: string;
+}
+
+/**
+ * Button color
+ */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+/**
+ * Component for AddEntry
+ * @requires values
+ * 
+ * @example 
+ *             <AddEntry
+ *             />
+ */
+
+class AddEntry extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: PropsType) {
+    super(props);
+    this.state = {
+      volume: '0',
+      quality: '0',
+      costPerUnit: '0',
+    };
+  }
+
+  /** Get current datetime in specified format */
+  private getDatetime = (format: string) => moment().format(format);
+
+  /** Create page buttons */
+  private renderCancelButton = () => this.renderButton('Cancel', 'INFO', this.onCancelPress);
+  private renderSaveButton = () => this.renderButton('Save', 'PRIMARY', this.onSavePress);
+
+  /** Handle pressing cancel button */
+  private onCancelPress = () => this.props.goBack();
+  
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    const timeNow = moment().utc().toString();
+    let newEntry: MilkEntry = {
+      uuid: 'fakeEntryUuid',
+      toUUID: 'fakeToUuid',
+      fromUUID: 'fakeFromUuid',
+      datetime: timeNow,
```

Comment:
The default `toString` method of moment is probably not going to be the same format Enoch wants it on the backend. We have to match for format `YYYY-MM-DD hh-mm-ss` for anything going to the backend (which will hopefully be validated on the core at some point). I don't know if my capitalization is correct on the various fields btw so don't just stick that in assuming I know what I'm doing :P

Reply:
@jeinglis @enochtsang We'll have to be in sync with this one.